code/frontend/MakeItSo/Shared/Features/ReminderList/Views/ReminderDateTimeView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//	ReminderDateTimeView.swift
+//  MakeItSo
+//
+//  Created by Andrew Cowley on 08/12/2021.
+//  Copyright © 2021 Google LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import SwiftUI
+
+// Make the date time toggle section in the ReminderDetailsView
+struct ReminderDateTimeView: View {
+  @EnvironmentObject var viewModel: ReminderDetailsViewModel
+
+  var body: some View {
+    Toggle(isOn: $viewModel.hasDueDate) {
+      dateToggleLabel
+    }
+    .animation(.none, value: viewModel.pickerState )
+
+    if viewModel.pickerState == .date {
+      DatePicker("Date", selection: $viewModel.dueDate, displayedComponents: .date)
+        .datePickerStyle(.graphical)
+    }
+
+    Toggle(isOn: $viewModel.hasDueTime) {
+      timeToggleLabel
+    }
+    if viewModel.pickerState == .time {
+      DatePicker("Time", selection: $viewModel.dueDate, displayedComponents: .hourAndMinute)
+        .datePickerStyle(.wheel)
+    }
+  }
+
+  // Toggle label subviews for body
+  var dateToggleLabel: some View {
```

Comment:
Can probably refactor this even more by turning into a function and passing in the icon and title.

Rename the function to something like `expandableRow` or `expandableRowLabel`, and replace `timeToggleLable` entirely.

Reply:
I thought about it, but you would to pass five parameters actually:
icon, title, dateValue, handler, textStyle

 HStack {
      Image(icon)
        .frame(width: 26, height: 26) //, alignment: .center)
        .background(.blue)
        .foregroundColor(.white)
        .cornerRadius(4)
      VStack(alignment: .leading) {
        Text(title)
        if dateValue != nil {
          Button(action: handler) {
            Text(dateValue, style: style)
        } ...

Not worth it for something that likely won't be re-used again?